frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -3,44 +3,69 @@ import * as _ from 'lodash';
 import * as classNames from 'classnames';
 import {
   FieldLevelHelp,
-  RequestSizeInput,
   withHandlePromise,
+  humanizeBinaryBytes,
 } from '@console/internal/components/utils/index';
 import {
   createModalLauncher,
   ModalBody,
   ModalSubmitFooter,
   ModalTitle,
 } from '@console/internal/components/factory';
-import { k8sPatch, K8sResourceKind } from '@console/internal/module/k8s';
+import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
+import { k8sPatch } from '@console/internal/module/k8s';
+import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
 import { OCSServiceModel } from '../../../models';
+import { OSD_CAPACITY_SIZES } from '../../../utils/osd-size-dropdown';
+import { CEPH_STORAGE_NAMESPACE } from '../../../constants';
 import './_add-capacity-modal.scss';
-import { OCSStorageClassDropdown } from '../storage-class-dropdown';
+
+const labelTooltip =
+  'The backing storage requested will be higher as it will factor in the requested capacity, replica factor, and fault tolerant costs associated with the requested capacity.';
 
 export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps) => {
   const { ocsConfig, close, cancel } = props;
-  const dropdownUnits = {
-    TiB: 'TiB',
-  };
-  const requestSizeUnit = dropdownUnits.TiB;
-  const [buttonDisabled, setButton] = React.useState(false);
-  const [requestSizeValue, setRequestSizeValue] = React.useState('2');
-  const [storageClass, setStorageClass] = React.useState('');
+  const osdSizeWithUnit = _.get(
+    ocsConfig,
+    'spec.storageDeviceSets[0].dataPVCTemplate.spec.resources.requests.storage',
+  );
+  const osdSizeWithoutUnit = OSD_CAPACITY_SIZES[osdSizeWithUnit].size;
+
+  const [requestSizeValue, setRequestSizeValue] = React.useState(osdSizeWithoutUnit);
+  const [response, loadError, loading] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY,
+    namespace: CEPH_STORAGE_NAMESPACE,
+    query: 'ceph_cluster_total_used_bytes',
```

Comment:
import this from the query file

Reply:
I need to then import three objects for that.

Reply:
Done